assets/js/phoenix.js

```diff
@@ -966,9 +984,22 @@ export class Socket {
    * @param {Channel}
    */
   remove(channel){
+    this.off(channel)
     this.channels = this.channels.filter(c => c.joinRef() !== channel.joinRef())
   }
 
+  /**
+   * stateChangeCallbacks is an array of "tuples" [[ref, callback], [...]]
+   * and can be cleaned up when a channel goes offline
+   * @param {Channel}
+   */
+  off(channel) {
+    let refs = channel.stateChangeRefs
+    this.stateChangeCallbacks.open = this.stateChangeCallbacks.open.filter(([ref]) => {
+      return !refs.includes(ref)
+    })
```

Comment:
I'm a follow up (or current) addition to this PR is cleaning up other containers for callbacks like `error`...